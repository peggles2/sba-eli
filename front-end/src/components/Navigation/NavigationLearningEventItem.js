import React, { Component } from "react";
import { Dropdown } from "semantic-ui-react";

export default class NavigationLearningEventItem extends Component {
  render() {
    return (
      <Dropdown.Item href={this.props.path} text={this.props.name} />
    );
  }
}
