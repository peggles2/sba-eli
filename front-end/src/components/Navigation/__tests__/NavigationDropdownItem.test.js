import React from 'react';
import ReactDOM from 'react-dom';
import NavigationLearningEventItem from '../NavigationLearningEventItem'
import { shallow, mount } from 'enzyme';

describe('NavigationLearningEventItem', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavigationLearningEventItem/>);
    
    expect(wrapper).toMatchSnapshot();
  });

});
