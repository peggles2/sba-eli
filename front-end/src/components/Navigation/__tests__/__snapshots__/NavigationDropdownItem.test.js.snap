// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationDropdownItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationDropdownItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "additionLabel": "Add ",
      "additionPosition": "top",
      "children": <DropdownMenu>
        <DropdownItem
          as={[Function]}
        />
      </DropdownMenu>,
      "closeOnBlur": true,
      "deburr": false,
      "icon": "dropdown",
      "item": true,
      "minCharacters": 1,
      "noResultsMessage": "No results found.",
      "openOnFocus": true,
      "renderLabel": [Function],
      "searchInput": "text",
      "selectOnBlur": true,
      "selectOnNavigation": true,
      "text": undefined,
      "wrapSelection": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <DropdownItem
          as={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": [Function],
          "text": undefined,
          "to": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "additionLabel": "Add ",
        "additionPosition": "top",
        "children": <DropdownMenu>
          <DropdownItem
            as={[Function]}
          />
        </DropdownMenu>,
        "closeOnBlur": true,
        "deburr": false,
        "icon": "dropdown",
        "item": true,
        "minCharacters": 1,
        "noResultsMessage": "No results found.",
        "openOnFocus": true,
        "renderLabel": [Function],
        "searchInput": "text",
        "selectOnBlur": true,
        "selectOnNavigation": true,
        "text": undefined,
        "wrapSelection": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <DropdownItem
            as={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "text": undefined,
            "to": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
